src/rust/generator-dispatcher/tests/test_utils/context.rs

```diff
@@ -0,0 +1,91 @@
+use std::time::Duration;
+
+use clap::Parser;
+use opentelemetry::{
+    global,
+    sdk::propagation::TraceContextPropagator,
+};
+use plugin_work_queue::{
+    psql_queue::PsqlQueue,
+    PluginWorkQueueDbConfig,
+};
+use rust_proto::{
+    graplinc::grapl::api::pipeline_ingress::v1beta1::client::PipelineIngressClient,
+    protocol::{
+        healthcheck::client::HealthcheckClient,
+        service_client::NamedService,
+    },
+};
+use test_context::AsyncTestContext;
+use tracing_appender::non_blocking::WorkerGuard;
+use tracing_subscriber::{
+    prelude::*,
+    EnvFilter,
+};
+
+pub struct GeneratorDispatcherTestContext {
+    pub pipeline_ingress_client: PipelineIngressClient,
+    pub plugin_work_queue_psql_client: PsqlQueue,
+    pub _guard: WorkerGuard,
+}
+
+#[async_trait::async_trait]
+impl AsyncTestContext for GeneratorDispatcherTestContext {
+    async fn setup() -> Self {
+        let (non_blocking, _guard) = tracing_appender::non_blocking(std::io::stdout());
+
+        // initialize json logging layer
+        let log_layer = tracing_subscriber::fmt::layer()
+            .json()
+            .with_writer(non_blocking);
+
+        // initialize tracing layer
+        global::set_text_map_propagator(TraceContextPropagator::new());
+        let tracer = opentelemetry_jaeger::new_pipeline()
+            .with_service_name("generator-dispatcher-integration-tests")
+            .install_batch(opentelemetry::runtime::Tokio)
+            .expect("could not configure tracer");
+
+        // register a subscriber
+        let filter = EnvFilter::from_default_env();
+        tracing_subscriber::registry()
+            .with(filter)
+            .with(log_layer)
+            .with(tracing_opentelemetry::layer().with_tracer(tracer))
+            .init();
+
```

Comment:
I think adding a single tracing subscriber should be done using the `test-log` crate with `features=["trace"]`

Reply:
If we do that here the log messages in this method won't be logged.. I think? Anyway, I'll defer to @wimax-grapl for this one as well